pkg/client/config/config.go

```diff
@@ -0,0 +1,189 @@
+type PublicRestConfigClient struct {
+	URL  *common.URL       `json:"url" yaml:"url"`
+	Auth *PublicAuthConfig `json:"auth,omitempty" yaml:"auth,omitempty"`
+	// The HTTP authorization credentials for the targets.
+	Authorization *secret.PublicAuthorization `json:"authorization,omitempty" yaml:"authorization,omitempty"`
+	// TLSConfig to use to connect to the targets.
+	TLSConfig *secret.PublicTLSConfig `json:"tls_config,omitempty" yaml:"tls_config,omitempty"`
+	Headers   map[string]string       `json:"headers,omitempty" yaml:"headers,omitempty"`
+}
+
+func NewPublicRestConfigClient(config *RestConfigClient) *PublicRestConfigClient {
+	if config == nil {
+		return nil
+	}
+	return &PublicRestConfigClient{
+		URL:           config.URL,
+		Auth:          newPublicAuthConfig(config.Auth),
+		Authorization: secret.NewPublicAuthorization(config.Authorization),
+		TLSConfig:     secret.NewPublicTLSConfig(config.TLSConfig),
+		Headers:       config.Headers,
+	}
+}
+
+type OauthConfig struct {
+	// ClientID is the application's ID.
+	ClientID string `json:"client_id" yaml:"client_id"`
+	// ClientSecret is the application's secret.
+	ClientSecret string `json:"client_secret" yaml:"client_secret"`
+	// TokenURL is the resource server's token endpoint
+	// URL. This is a constant specific to each server.
+	TokenURL string `json:"token_url" yaml:"token_url"`
+	// Scope specifies optional requested permissions.
+	Scopes []string `json:"scopes" yaml:"scopes"`
+	// EndpointParams specifies additional parameters for requests to the token endpoint.
+	EndpointParams url.Values `json:"endpoint_params" yaml:"endpoint_params"`
+	// AuthStyle optionally specifies how the endpoint wants the
+	// client ID & client secret sent. The zero value means to
+	// auto-detect.
+	AuthStyle oauth2.AuthStyle `json:"auth_style" yaml:"auth_style"`
+}
+
+type AuthConfig struct {
+	NativeAuth  *api.Auth         `json:"native_auth,omitempty" yaml:"native_auth,omitempty"`
+	OauthConfig *OauthConfig      `json:"oauth_config,omitempty" yaml:"oauth_config,omitempty"`
+	BasicAuth   *secret.BasicAuth `json:"basic_auth,omitempty" yaml:"basic_auth,omitempty"`
+}
+
+// RestConfigClient defines all parameters that can be set to customize the RESTClient
+type RestConfigClient struct {
+	URL  *common.URL `json:"url" yaml:"url"`
+	Auth *AuthConfig `json:"auth,omitempty" yaml:"auth,omitempty"`
+	// The HTTP authorization credentials for the targets.
+	Authorization *secret.Authorization `json:"authorization,omitempty" yaml:"authorization,omitempty"`
+	// TLSConfig to use to connect to the targets.
+	TLSConfig *secret.TLSConfig `json:"tls_config,omitempty" yaml:"tls_config,omitempty"`
+	Headers   map[string]string `json:"headers,omitempty" yaml:"headers,omitempty"`
+}
+
+// NewFromConfig create an instance of RESTClient using the config passed as parameter
+func NewFromConfig(config RestConfigClient) (*perseshttp.RESTClient, error) {
```

Comment:
Not possible to have the client ctor inside the perseshttp package? 
To me it should be either ``perseshttp.NewFromConfig``, or eventually ``config.NewRESTClient``

Reply:
Ih had to do this to avoid a circular dependencies.

Yeah I think renaming it `config.NewRESTClient` makes sense. Thanks